pkg/datasource/events.go

```diff
@@ -387,6 +434,138 @@ func dataFrameForEventType(assets []schemas.Asset, eventType schemas.EventType,
+		for i := range events {
+			if events[i].Parent == nil {
+				continue
+			}
+
+			parentEvent := *events[i].Parent
+			parentEventType, parentEventTypeExists := eventTypes[parentEvent.EventTypeUUID]
+			if !parentEventTypeExists {
+				continue
+			}
+
+			parentPrefix := parentEventType.Name + "_"
+			if !parentFieldsAdded[parentPrefix] {
+				parentFieldsAdded[parentPrefix] = true
+
+				// Add parent fields
+				fieldByColumn[parentPrefix+EventUUIDColumnName] = data.NewField(parentPrefix+EventUUIDColumnName, nil, []*string{})
+				fieldByColumn[parentPrefix+StartTimeColumnName] = data.NewField(parentPrefix+StartTimeColumnName, nil, []*time.Time{})
+				fieldByColumn[parentPrefix+StopTimeColumnName] = data.NewField(parentPrefix+StopTimeColumnName, nil, []*time.Time{})
+				fieldByColumn[parentPrefix+DurationColumnName] = data.NewField(parentPrefix+DurationColumnName, nil, []*float64{})
+				fieldByColumn[parentPrefix+AssetUUIDColumnName] = data.NewField(parentPrefix+AssetUUIDColumnName, nil, []*string{})
+				fieldByColumn[parentPrefix+AssetColumnName] = data.NewField(parentPrefix+AssetColumnName, nil, []*string{})
+				fieldByColumn[parentPrefix+AssetPathColumnName] = data.NewField(parentPrefix+AssetPathColumnName, nil, []*string{})
+				fieldByColumn[parentPrefix+EventTypeUUIDColumnName] = data.NewField(parentPrefix+EventTypeUUIDColumnName, nil, []*string{})
+				fieldByColumn[parentPrefix+EventTypeColumnName] = data.NewField(parentPrefix+EventTypeColumnName, nil, []*string{})
+				parentFields = append(parentFields, fieldByColumn[parentPrefix+EventUUIDColumnName], fieldByColumn[parentPrefix+StartTimeColumnName], fieldByColumn[parentPrefix+StopTimeColumnName], fieldByColumn[parentPrefix+DurationColumnName], fieldByColumn[parentPrefix+AssetUUIDColumnName], fieldByColumn[parentPrefix+AssetColumnName], fieldByColumn[parentPrefix+AssetPathColumnName], fieldByColumn[parentPrefix+EventTypeUUIDColumnName], fieldByColumn[parentPrefix+EventTypeColumnName])
+
+				for _, parentEventTypeProperty := range eventTypePropertiesForEventType[parentEvent.EventTypeUUID] {
+					if parentEventTypeProperty.Type == schemas.EventTypePropertyTypePeriodic {
+						continue
+					}
+
+					parentPropertyFieldName := parentPrefix + parentEventTypeProperty.Name
+					var parentField *data.Field
+					switch parentEventTypeProperty.Datatype {
+					case schemas.EventTypePropertyDatatypeBool:
+						parentField = data.NewField(parentPropertyFieldName, nil, []*bool{})
+					case schemas.EventTypePropertyDatatypeNumber:
+						parentField = data.NewField(parentPropertyFieldName, nil, []*float64{})
+					case schemas.EventTypePropertyDatatypeString:
+						parentField = data.NewField(parentPropertyFieldName, nil, []*string{})
+					default:
+						parentField = data.NewField(parentPropertyFieldName, nil, []interface{}{})
+					}
+					parentFields = append(parentFields, parentField)
+					fieldByColumn[parentPropertyFieldName] = parentField
+				}
+			}
```

Comment:
we can also move this piece in the first iteration of the events on line 356... this spares us at least an iteration

Reply:
take a look here:   feature/sub-events-davis